### PR TITLE
Introduce a style guide to the docs

### PR DESCRIPTION
**Pull request by settermjd**
As the first step in seeking to ensure a consistent quality across the documentation, this PR introduces a very early version style guide, which sets out how the docs should be written.

### CODE REVIEW

File: docs/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- STDOUT
+- Traefik
+- localhost
+- webhooks
+- OpenSSL
+- RPC
+- URL
+- GDPR
+
+[the AsciiDoc recommended practices]: https://asciidoctor.org/docs/asciidoc-recommended-practices/
```
Comment (mmattel) [inline docs/style-guide.md]: You may also want to link our best practices guide,
because you have to imlement bp with attributes and directives :smile:

Comment (settermjd) [reply docs/style-guide.md]: What's "bp"?

Comment (settermjd) [reply docs/style-guide.md]: On reviewing the best practices guide, I don't see a natural way to link them. I'm open to suggestions though.

Comment (mmattel) [reply docs/style-guide.md]: as an external link: https://github.com/owncloud/docs/blob/master/docs/best-practices.md

File: docs/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Style Guide
+
+When contributing to the documentation, please ensure you follow [the AsciiDoc recommended practices].
+They are not currently rigidly enforced.
+However, this may change in the future.
+
+## Key Terms, Jargon, and Terminology
```
Comment (mmattel) [inline docs/style-guide.md]: What is the intention of having that in a style guide?

Comment (settermjd) [reply docs/style-guide.md]: It's a common attribute across a number of style guides. It shows how terms must be spelled, which avoids any ambiguity.

File: docs/style-guide.md (hunk 1 above)
Comment (settermjd) [reply docs/style-guide.md]: Thanks for the suggestion. I've updated the previous commit to include it.

**Comment by michaelstingl** [comment]
Related issue: https://github.com/owncloud/docs/issues/95

**Pull request merged**